javaslang/src/main/java/javaslang/collection/AbstractMap.java

```diff
@@ -21,238 +21,238 @@
  *
  * @param <K> Key type
  * @param <V> Value type
- * @param <M> Map type
+ * @param <R> Map type
  * @author Ruslan Sennov
  * @since 2.0.0
  */
-abstract class AbstractMap<K, V, M extends AbstractMap<K, V, M>> implements Map<K, V> {
+abstract class AbstractMap<K, V, R extends AbstractMap<K, V, R>> implements Map<K, V> {
```

Comment:
**M** = **M**ap
**R** = **R**esult

we will probably need it for non-maps also (e.g. `Queues`)